### DISCUSSION TITLE
Alpine Expression Error: pgToggleable is not defined

### DISCUSSION THREAD

**Discussion by gallomaf** [Power-Components/livewire-powergrid#1383] [Q&A]
Hello Everyone,
I've spent a lot of hours on this error.

I'm updating livewire v2 to livewire v3 and powegrid from v3 to v5.3.5
it seems that other functions are working, but toggleable coloumns are not working and I have this errore:

`livewire.js?id=a27c4ca2:1235 Alpine Expression Error: pgToggleable is not defined

Expression: "pgToggleable(JSON.parse('{\u0022id\u0022:120,\u0022isHidden\u0022:\u0022false\u0022,\u0022tableName\u0022:\u0022default\u0022,\u0022field\u0022:\u0022invio_provvigioni\u0022,\u0022toggle\u0022:0,\u0022trueValue\u0022:\u0022Yes\u0022,\u0022falseValue\u0022:\u0022No\u0022}'))"

 <div x-data=​"pgToggleable(JSON.parse('{\u0022id\u0022:​120,\u0022isHidden\u0022:​\u0022false\u0022,\u0022tableName\u0022:​\u0022default\u0022,\u0022field\u0022:​\u0022invio_provvigioni\u0022,\u0022toggle\u0022:​0,\u0022trueValue\u0022:​\u0022Yes\u0022,\u0022falseValue\u0022:​\u0022No\u0022}​')​)​">​<div class=​"flex">​flex<div :class=​"{
                    'relative rounded-full w-8 h-4 transition duration-200 ease-linear':​ true,
                    'bg-pg-secondary-600 dark:​pg-secondary-500':​ toggle,
                    'bg-pg-primary-200':​ !toggle
                }​" class>​<label :class=​"{
                        'absolute left-0 bg-white border-2 mb-2 w-4 h-4 rounded-full transition transform duration-100 ease-linear cursor-pointer':​ true,
                        'translate-x-full border-pg-secondary-600':​ toggle,
                        'translate-x-0 border-pg-primary-200':​ !toggle
                    }​" x-on:click=​"save" class>​</label>​<input type=​"checkbox" class=​"appearance-none opacity-0 w-full h-full active:​outline-none focus:​outline-none" x-on:click=​"save">​</div>​</div>​</div>​
handleError @ livewire.js?id=a27c4ca2:1235
(anonime) @ livewire.js?id=a27c4ca2:1308
Promise.catch (asinc)
(anonime) @ livewire.js?id=a27c4ca2:1308
tryCatch @ livewire.js?id=a27c4ca2:1228
evaluate @ livewire.js?id=a27c4ca2:1252
(anonime) @ livewire.js?id=a27c4ca2:3502
flushHandlers @ livewire.js?id=a27c4ca2:1390
stopDeferring @ livewire.js?id=a27c4ca2:1395
deferHandlingDirectives @ livewire.js?id=a27c4ca2:1398
initTree @ livewire.js?id=a27c4ca2:913
(anonime) @ livewire.js?id=a27c4ca2:868
start @ livewire.js?id=a27c4ca2:867
start2 @ livewire.js?id=a27c4ca2:8399
(anonime) @ livewire.js?id=a27c4ca2:9579
livewire.js?id=a27c4ca2:1235 Alpine Expression Error: toggle is not defined

Expression: "{
                    'relative rounded-full w-8 h-4 transition duration-200 ease-linear': true,
                    'bg-pg-secondary-600 dark:pg-secondary-500': toggle,
                    'bg-pg-primary-200': !toggle
                }"

 <div :class=​"{
                    'relative rounded-full w-8 h-4 transition duration-200 ease-linear':​ true,
                    'bg-pg-secondary-600 dark:​pg-secondary-500':​ toggle,
                    'bg-pg-primary-200':​ !toggle
                }​" class>​…​</div>​
handleError @ livewire.js?id=a27c4ca2:1235
(anonime) @ livewire.js?id=a27c4ca2:1308
Promise.catch (asinc)
(anonime) @ livewire.js?id=a27c4ca2:1308
tryCatch @ livewire.js?id=a27c4ca2:1228
(anonime) @ livewire.js?id=a27c4ca2:3475
reactiveEffect @ livewire.js?id=a27c4ca2:2402
effect2 @ livewire.js?id=a27c4ca2:2377
effect @ livewire.js?id=a27c4ca2:771
wrappedEffect @ livewire.js?id=a27c4ca2:787
handler2 @ livewire.js?id=a27c4ca2:3475
flushHandlers @ livewire.js?id=a27c4ca2:1390
stopDeferring @ livewire.js?id=a27c4ca2:1395
deferHandlingDirectives @ livewire.js?id=a27c4ca2:1398
initTree @ livewire.js?id=a27c4ca2:913
(anonime) @ livewire.js?id=a27c4ca2:868
start @ livewire.js?id=a27c4ca2:867
start2 @ livewire.js?id=a27c4ca2:8399
(anonime) @ livewire.js?id=a27c4ca2:9579
livewire.js?id=a27c4ca2:1235 Alpine Expression Error: toggle is not defined`
and

`livewire.js?id=a27c4ca2:1235 Alpine Expression Error: toggle is not defined

Expression: "{
                        'absolute left-0 bg-white border-2 mb-2 w-4 h-4 rounded-full transition transform duration-100 ease-linear cursor-pointer': true,
                        'translate-x-full border-pg-secondary-600': toggle,
                        'translate-x-0 border-pg-primary-200': !toggle
                    }"

 <label :class=​"{
                        'absolute left-0 bg-white border-2 mb-2 w-4 h-4 rounded-full transition transform duration-100 ease-linear cursor-pointer':​ true,
                        'translate-x-full border-pg-secondary-600':​ toggle,
                        'translate-x-0 border-pg-primary-200':​ !toggle
                    }​" x-on:click=​"save" class>​</label>​
handleError @ livewire.js?id=a27c4ca2:1235
(anonime) @ livewire.js?id=a27c4ca2:1308
Promise.catch (asinc)
(anonime) @ livewire.js?id=a27c4ca2:1308
tryCatch @ livewire.js?id=a27c4ca2:1228
(anonime) @ livewire.js?id=a27c4ca2:3475
reactiveEffect @ livewire.js?id=a27c4ca2:2402
effect2 @ livewire.js?id=a27c4ca2:2377
effect @ livewire.js?id=a27c4ca2:771
wrappedEffect @ livewire.js?id=a27c4ca2:787
handler2 @ livewire.js?id=a27c4ca2:3475
flushHandlers @ livewire.js?id=a27c4ca2:1390
stopDeferring @ livewire.js?id=a27c4ca2:1395
deferHandlingDirectives @ livewire.js?id=a27c4ca2:1398
initTree @ livewire.js?id=a27c4ca2:913
(anonime) @ livewire.js?id=a27c4ca2:868
start @ livewire.js?id=a27c4ca2:867
start2 @ livewire.js?id=a27c4ca2:8399
(anonime) @ livewire.js?id=a27c4ca2:9579
livewire.js?id=a27c4ca2:1235 Alpine Expression Error: pgToggleable is not defined`

Entire log: 
[127.0.0.1-1707059332823.log](https://github.com/Power-Components/livewire-powergrid/files/14157293/127.0.0.1-1707059332823.log)

Could help me to solve this issue? 
Thank you in advance
Francesco

**Comment by technoholicpl** [comment]
Have you tried this?

resources/js/app.js

`import './../../vendor/power-components/livewire-powergrid/dist/powergrid`'
